search-commons/src/main/java/no/unit/nva/search2/ticket/TicketQuery.java

```diff
@@ -158,26 +188,31 @@ public TicketQuery withUser(String username) {
      * @param ticketTypes the required types
      * @return TicketQuery (builder pattern)
      */
-    public TicketQuery withRequiredTicketType(TicketType... ticketTypes) {
-        var ticketStringTypes = Arrays.stream(ticketTypes).map(Object::toString).toList();
-        final var filter = new TermsQueryBuilder(TYPE_KEYWORD, ticketStringTypes)
-            .queryName(TICKETS + TYPE);
+    public TicketQuery withFilterTicketType(TicketType... ticketTypes) {
+        this.ticketTypes = ticketTypes.clone();
+        final var filter =
+            new TermsQueryBuilder(TYPE_KEYWORD, Arrays.stream(ticketTypes).map(TicketType::toString).toList())
+                .queryName(TICKETS + TYPE);
         this.addFilter(filter);
         return this;
     }
 
     /**
-     * Filter on organization.
-     * <P>Only documents belonging to organization specified are searchable (for the user)
-     * </p>
+     * Filter on owner (user).
      *
-     * @param organization uri of publisher
-     * @return ResourceQuery (builder pattern)
+     * <p>Only tickets owned by user will be available for the Query.</p>
+     * <p>This is to avoid the Query to return documents that are not available for the user.</p>
+     *
+     * @param userName current user
+     * @return TicketQuery (builder pattern)
      */
-    public TicketQuery withRequiredOrganization(URI organization) {
-        final var filter = new TermQueryBuilder(ORGANIZATION_ID_KEYWORD, organization.toString())
-            .queryName(ORGANIZATION + ID);
-        this.addFilter(filter);
+    public TicketQuery withFilterCurrentUser(String userName) {
+        this.currentUser = userName;
+        if (isUserOnly(ticketTypes)) {
```

Comment:
isNotCurrator ?

Reply:
isGeneralSupportCase?

Reply:
I am guessing what kind of user based on accessrights...